[Progresst-8] iP

Progresst-8 opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Progresst-8 committed:
Add level-0 increments Rename, Greet, Exit

Progresst-8 committed:
Add Level 1 Echo

Add personality to Yapper's text

Progresst-8 committed:
Add Level 2 Add, List

Add ability to store user input and output all of them when requested.
Add line divider strings

Progresst-8 committed:
Add Level-3 Mark as Done; Made program obey SRP

Added the ability to mark Tasks as Done or Undone;
Split Yapper into more classes to handle one responsibility each:
1 Yapper initializes TaskManager and runs the UserInterface.
2 UserInterface chooses the instruction based on user input.
3 TaskManager handles the various tasks stored by the user.
4 InputStringHandler parses strings inputted by the user.
5 OutputStringHandler formats strings to output to the user.
6 Instruction handles the various instructions (implemented so far).
7 Task contains features of tasks such as description is done-ness.
8 CommonStrings contains important strings used in the program.

Progresst-8 committed:
Add Level 4 ToDos, Events, Deadlines

Modified code to include the 3 new Task subtypes:
Add taskToString() for all of Task, TaskTodo, TaskDeadline and TaskEvent
InputStringHandler can now parse the 3 new instructions
OutputStringHandler can now format the 3 new instructions
Instruction now includes the 3 new instruction types
UserInterface now handles the 3 new instruction types

Progresst-8 committed:
Renamed class that stores ChatBot speeches from CommonStrings to StringStorage.

The old name was not representative of what it currently does.
Also modified the messages sent upon startup and shutdown, as I felt the old one was bad. The new one sounds better.

nirala-ts commented on src/main/java/InputStringHandler.java:
```diff
@@ -0,0 +1,37 @@
+        case "bye":
+            return new Instruction(Instruction.InstructionType.BYE);
+        }
+        return new Instruction(null); // shouldn't happen
```
Returning an Instruction with a null type is not ideal although you have commented that it would not happen. Consider throwing an exception or returning an error instruction type if the command is unrecognised. This will help avoid potential NullPointerException or unexpected behaviour.

nirala-ts commented on src/main/java/StringStorage.java:
```diff
@@ -0,0 +1,14 @@
+                                               + "Call for me whenever ya feel like listening again.\n"
+                                               + "Cya!"
+                                               + LINE_DIVIDER;
+}
```
LGTM!

nirala-ts commented on src/main/java/Instruction.java:
```diff
@@ -0,0 +1,46 @@
+        this.instructionType = type; // ADD, TODO
+        this.instructionStringArg = arg;
+    }
+    public Instruction(InstructionType type, String arg1, String arg2) {
```
It would probably be more intuitive to to rename arg1 and arg2 to from and to, or something more descriptive.

nirala-ts commented on src/main/java/Instruction.java:
```diff
@@ -0,0 +1,46 @@
+        this.instructionStringArg = arg1;
+        this.instructionStringArgs = new String[] {arg2};
+    }
+    public Instruction(InstructionType type, String arg1, String arg2, String arg3) {
```
Same thing, it would probably be more intuitive to to rename arg1, arg2 and arg 3 to something more descriptive.

nirala-ts commented on src/main/java/OutputStringHandler.java:
```diff
@@ -0,0 +1,52 @@
+    // For Command: Add
+    public static void printAddedTask(int taskCount, Task task) {
+        System.out.println(StringStorage.LINE_DIVIDER);
+        System.out.println("Now, ya gotta do this too:");
+        System.out.println( "  " + task.taskToString() );
+        System.out.println("Your list is now THIS BIG: " + taskCount);
+        System.out.println(StringStorage.LINE_DIVIDER);
+    }
+    // For Command: Todo
+    public static void printAddedTask(int taskCount, TaskTodo todo) {
+        System.out.println(StringStorage.LINE_DIVIDER);
+        System.out.println("Now, ya gotta do this too:");
+        System.out.println( todo.taskToString() );
+        System.out.println("Your list is now THIS BIG: " + taskCount);
+        System.out.println(StringStorage.LINE_DIVIDER);
+    }
+    // For Command: Deadline
+    public static void printAddedTask(int taskCount, TaskDeadline taskDeadline) {
+        System.out.println(StringStorage.LINE_DIVIDER);
+        System.out.println("Now, ya gotta do this too:");
+        System.out.println( taskDeadline.taskToString() );
+        System.out.println("Your list is now THIS BIG: " + taskCount);
+        System.out.println(StringStorage.LINE_DIVIDER);
+    }
+    // For Command: Event
+    public static void printAddedTask(int taskCount, TaskEvent taskEvent) {
+        System.out.println(StringStorage.LINE_DIVIDER);
+        System.out.println("Now, ya gotta do this too:");
+        System.out.println( taskEvent.taskToString() );
+        System.out.println("Your list is now THIS BIG: " + taskCount);
+        System.out.println(StringStorage.LINE_DIVIDER);
```
An alternative design would be to move the responsibility of outputting task-specific messages into the respective task classes (Task, TaskTodo, TaskDeadline, TaskEvent). This would also ensure that the class definition is not too long.

telferang replied:
Consider extracting "Now, ya gotta do this too:" and "Your list is now THIS BIG: " into a constant to make the code more readable and easier to modify.

nirala-ts reviewed (commented):
Overall, I found your code easy to read for the most part except for a few places
where variable names were not intuitive. Do Consider splitting into
smaller classes as well.

telferang commented on src/main/java/UserInterface.java:
```diff
@@ -0,0 +1,81 @@
+        OutputStringHandler.printTaskStatus(task, true);
+    }
+    public static void handleUnmarkInstruction(TaskManager taskManager, Integer taskOrdinal) {
+        Task task = taskManager.getTask(taskOrdinal - 1); // 0-Indexed
```
Consider using a constant such as INDEX_OFFSET instead of using the magic number 1 for indexing adjustment. This will improve readability and make future adjustments easier.

telferang commented on src/main/java/UserInterface.java:
```diff
@@ -0,0 +1,81 @@
+                System.out.println(StringStorage.SHUT_DOWN_MESSAGE);
+                return;
+            }
+            // TBA ?
```
The code does not seem to handle case where user input is invalid. Consider adding error handling to improve robustness and user experience.

telferang commented on src/main/java/StringStorage.java:
```diff
@@ -0,0 +1,14 @@
+// Contains strings for the Yapper program
+public class StringStorage {
+    public static final String LINE_DIVIDER = "____________________________________________________________";
```
Extracting the LINE_DIVIDER as a public static final is good since it ensures that this constant is immutable and easily accessible across the program.

telferang commented on src/main/java/Yapper.java:
```diff
@@ -0,0 +1,9 @@
+// Main function for Yapper
+public class Yapper {
+    public static void main(String[] args) {
+        // Initialize
+        TaskManager taskManager = new TaskManager(100);
+        // Startup ChatBot Program
+        UserInterface.startYappin(taskManager);
+    }
+}
```
LGTM

telferang reviewed (commented):
Overall a good job using SLAM. There are a few minor changes you can consider

Progresst-8 committed:
Amended abbr for Todo Deadline Event when printed to more closely align to the CS2113 website

Progresst-8 committed:
Amended Constructor arguments to be more descriptive

Progresst-8 committed:
Extracted 2 strings into a constant

For code readability and easier modification later on

Progresst-8 committed:
Moved the print methods to the respective task classes

This shortens the definition for class OutputStringHandler

Progresst-8 committed:
Renamed variable to make clearer what it is

Progresst-8 committed:
Fix OOB error caused when making deadline or event

Progresst-8 committed:
Add error handling and refactor strings

Progresst-8 committed:
Refactor variables and magic numbers

Progresst-8 committed:
Add a class for error handling

description empty or empty input error is handled

Progresst-8 committed:
Add tentative method definitions for error handling

Progresst-8 committed:
Package classes, removed ADD from Instruction Set

Package entire program into one, also packaged the Task, Todo, Deadline, Event class into its own package/
add and todo used to do the same thing, so add instruction is subsumed under todo instruction.

Progresst-8 committed:
Merge branch 'master' into branch-Level-5

Progresst-8 committed:
refactor out main program loop into its own method

startYappin now consists of print statements and a loop, but no switch case statement
handleInstruction consists of a big switch case statement

Progresst-8 committed:
Moved exception, instruction classes into their own package

Exception classes include: ErrorHandler YapperException
Instruction classes include: InstructionHandler Instruction

Progresst-8 committed:
Rename children of Task

Progresst-8 committed:
Rename package stringHandlers to io

The old name felt too long

Progresst-8 committed:
Merge branch 'master' into branch-Level-5

Progresst-8 committed:
Add exception handling to Yapper program

A lot of throw-try-catch blocks added

Progresst-8 committed:
Update gitignore file

Progresst-8 committed:
Update gitignore file

Progresst-8 committed:
Update runtest to follow the new packaging

Progresst-8 committed:
Changed code layout of StringStorage for cleanliness

Progresst-8 committed:
Updated input.txt and expected.txt, shortened code for runtest.bat

Progresst-8 committed:
Simplified the code in various places

Removal of duplicated code:
    The various handleAddInstruction methods were consolidated into 1
    The various handle Mark and Unmark methods were consolidated into 1
Cleaning up of print methods:
    printAddedTask has been removed from Task children as it is currently unecessary
    printWithDividers to clearly separate input messages and output messages
Task children prefixes refactored into StringStorage
Exception handling added to parseUserInput
Various comments make to separate code more clearly

Progresst-8 committed:
Merge branch 'cleanUpCode'

Progresst-8 committed:
Renamed TaskManager to TaskHandler for consistency

Moved TaskHandler to task package

Progresst-8 committed:
Create jar manifest

Change EXPECTED.txt

Progresst-8 committed:
Update readme for Setup and User Guide

Progresst-8 committed:
Fix error in code that was storing event dates wrongly

end date was stored as both start and end date

Progresst-8 committed:
Update EXPECTED.TXT to have intended output

Progresst-8 committed:
Use Java Collections ArrayList<Task> instead of an array

\\

Progresst-8 committed:
Amend code to reflect using arrayList instead of array

Method parameters and element access changed to reflect new data structure,
No longer a need for maxCapacity since data structure is now dynamic and not static
No longer a need for checkIfListFull since data structure has no maxCapacity
No longer a need for try-catch in handleAddInstruction

Progresst-8 committed:
Amend code to reflect using arrayList instead of array

No longer a need for maxCapacity

Progresst-8 committed:
Add Delete to list of instructions

Removed BYE from InstructionType
Moved error handling from printTasks (in class OutputStringHandler) to handleListInstruction (in class InstructionHandler)
Changed code in handleMarkingInstruction
Add DELETE to parseUserInput (in InputStringHandler class)
Add DELETE to handleInstruction (in InstructionHandler class)
Add handleDeleteInstruction (in InstructionHandler class)
Add deleteTask (in TaskHandler class)
Add printDeletedTask (in Task class)
Add output message to StringStorage when task is deleted
Changed how catch blocks print to output

Progresst-8 committed:
Update UserGuide README

Progresst-8 committed:
Merge branch 'branch-Level-6'

Progresst-8 committed:
Add ability to save task list into file

Add storing and loading data methods in TaskHandler
Add various stringToTask and taskToString methods to the Task class and subclasses

Progresst-8 committed:
Resolve merge conflicts from branch 'branch-Level-7' between Task and TaskHandler

Add strings relating to saving and loading and corresponding error messages in StringStorage
Add error handling methods during saving and loading in ErrorHandler
Add save-data and load-data handling methods in InstructionHandler
Add taskToDisplay (for display during program run), taskToString (for saving data), stringToTask (for loading save data) method for each of Task, Todo, Deadline, Event
Add constructor that takes in boolean for task done status for each of Task, Todo, Deadline, Event
Add storeTaskData and loadTaskData methods in TaskHandler
Add stringToTasks method to select the correct Task subchild from string in TaskHandler

Progresst-8 committed:
Change stuff in user guide README

Progresst-8 committed:
Change startup, shutdown and help messages.

Changed the startYappin method accordingly

Progresst-8 committed:
Refactor symbols related to output and put them into StringStorage

Progresst-8 committed:
Made changes regarding error handling

Move error handling out of updateTaskStatus (in TaskHandler) into handleMarkingInstruction (in InstructionHandler)
Rename OOB check method as it was misnamed
Did some rearranging of methods in ErrorHandler
Add some comments

Progresst-8 committed:
Move file handling methods into its own class

Progresst-8 committed:
Add error handling when reading from user or file

Removed stringToTask methods (in Task Todo Deadline Event), now this responsibility is to parseFileInput (in SaveFileHandler class)
Add methods (in ErrorHandler class) to check for missing arguments for Todo Deadline List or if task type and status are valid
The abovementioned new methods for missing arguments are used in in parseFileInput method (in SaveFileHandler class) and parseUserInput method (in InputStringHandler class)
Add relavant error messages (in StringStorage class)

Progresst-8 committed:
Add input validation methods and some comments

Progresst-8 committed:
Move print methods out from TaskHandler and into InstructionHandler

Progresst-8 committed:
Refactor attribute and rename class

Refactor INDEX_OFFSET
Renamed ErrorHandler class to ExceptionHandler

Progresst-8 committed:
Rewrote methods in SaveFileHandler

Added those methods to InstructionHandler and Yapper

Progresst-8 committed:
Moved throwing exception to default switch case branch

Progresst-8 committed:
Remove WIP save and load methods as it is now implemented differently

Progresst-8 committed:
Move printAddedTask and printDeletedTask methods from Task to OSH

Progresst-8 committed:
Update method used to printing the while list to output

Progresst-8 committed:
Update desc of taskToString for Todo Deadline Event

Progresst-8 committed:
Rename checkIfListEmpty method to overload checkIfTaskOrdinalIsOutOfRange

also, Move print methods from Task to OutputStringHandler

Progresst-8 committed:
Move delimiter strings from InputStringHandler class to StringStorage class

Progresst-8 committed:
Extract prefix strings into a constant

Progresst-8 committed:
Move instruction prefixes from InputStringHandler to StringStorage

Progresst-8 committed:
Add comments, getters, setters

Progresst-8 committed:
Rework methods

In class SaveFileHandler, reworked methods loadTasks and loadTask to use ExceptionHandler directly
In class InputStringHandler, rework method parseUserInput to use ExceptionHandler directly
In class InputStringHandler, add methods splitStringByDeadlineKeyword and splitStringByEventKeyword
In class ExceptionHandler, combined 2 check-if-string-empty methods into a single checkIfUserInputStringEmpty method
In class ExceptionHandler, add methods checkIfDeadlineKeywordsPresent and checkIfEventKeywordsPresent
In class ExceptionHandler, changed some comments and strings

Progresst-8 committed:
Change excepion messages

Changed 3 DELIMITER strings to be public static final instead of static (I should've done this earlier)
Add PREFIX string for BYE instruction
Change some comments
Add isEmpty to TaskHandler

Progresst-8 committed:
Made some minor changes

Move .trim() code out of handleInstruction method (in InstructionHandler class) into parseUserInput method (in InputStringHandler class)
Change strings in SaveFileHandler
Change code in InputStringHandler
Format code in InputStringHandler
Change code to compare strings from using "string1 != string2" to using "!string1.equals(string2)" (in InputStringHandler class)
Change code that splits string into type and args from .indexOf() & .substring() to .split() (in InputStringHandler class)
Change code in checkIfUserInputEmpty method to differentiate between when userInput empty and when arguments are missing (in ExceptionHandler)

Progresst-8 committed:
Made exception message more descriptive

Deleted unused methods in ExceptionHandler class

Progresst-8 committed:
Correct error where task is marked as done when it isn't, and vice versa

Progresst-8 committed:
Add and modified exception handling methods

Some exception messages were refactored to StringStorage class
Add some newlines to strings in StringStorage class
Amend error in OutputStringHandler that was using taskToString instead of taskToDisplay

Progresst-8 committed:
Change the way line dividers are used in Yapper

Progresst-8 committed:
Amend error in OutputStringHandler that used taskToString instead of taskToDisplay

Progresst-8 committed:
Made changes in InstructionHandler class

Conversion of ordinals between 0-index (ArrayList) and 1-index (Yapper output) is moved from the individual handlers (delete, mark, unmark) to the main handler
Those individual handlers now throw to the main handler
Some exception handling strings were changed in the individual handlers
handleInstruction now has 2 try-catch blocks: one for parsing and one for executing

Progresst-8 committed:
Removed line dividers from OutputStringHandler methods

Due to accounting for possible exception messages after these print statements, the main Yapper class handles it instead

Progresst-8 committed:
Change code in SaveFileHandler class

Change string SAVE_FILE_PATH
Removed print statements (that check if there are errors during saving/loading process) as the try-catch blocks handle it
Removed ordinal correction as that is already handled by handleInstruction in an earlier commit
loadTask now has a default switch-case, so the return null is not needed.
Add code to create new txt file and folder if not already created.

Progresst-8 committed:
Change IS_DONE and NOT_DONE symbols to X and O

The old pair includes a " " which is may be misleading as to whether the savefile is corrupted or not

Progresst-8 committed:
Refactored out LINE_DIVIDER from if-else blocks

Progresst-8 committed:
Add limit for split to check for missing arguments

Progresst-8 committed:
Fix exception handling for file input-output

Add check-for-empty code in loadTasks method
Add check for fields in file for exception handling
Add a comma in YapperException message to make separate line in file and message
Add split-by-delimiter methods to SaveFileHandler
Add limit to .split() for exception handling purposes
Change taskToString methods for Task Todo Deadline Event
Comment out strings in StringStorage class
Add split-by-delimiter methods to InputStringHandler and amend relevant code in the method above
Rename comment in InputStringHandler

Progresst-8 committed:
Add exception handling if user input doesn't start with any of the known prefixes

Progresst-8 committed:
Split file handling class into input and output specific classes

Refactor SAVE_FILE_PATH from SaveFileHandler to StringStorage
Split SaveFileHandler class into classes InputFileHandler and OutputFileHandler

Progresst-8 committed:
Add JavaDoc comments to my code

Add JavaDoc comments to Yapper class, and its methods: main, startYappin
Add JavaDoc comments to ExceptionHandler class, and all its 12 methods
Add JavaDoc comments to YapperException class, and its methods: Constructor
Add JavaDoc comments to InputFileHandler class, and all its 3 methods
Add JavaDoc comments to OutputFileHandler class, and all its 4 methods
Add JavaDoc comments to InputStringHandler class, and all its 3 methods
Add JavaDoc comments to OutputStringHandler class, and all its 4 methods
Add JavaDoc comments to StringStorage class, and all its 2 methods: splitByDelimiter, printWithDividers. The various attributes/fields are grouped by purpose and those groups get a JavaDoc.
Add JavaDoc comments to Instruction class, InstructionType enum, all of its 4 attributes/fields and none of its methods
Add JavaDoc comments to InstructionHandler class, and all its 5 methods
Add JavaDoc comments to TaskHandler class, and 6 of its methods
Add JavaDoc comments to Task class, and 7 of its methods
Add JavaDoc comments to Todo class, and 2 of its methods
Add JavaDoc comments to Deadline class, and 2 of its methods
Add JavaDoc comments to Event class, and 2 of its methods

Progresst-8 committed:
Add date-related attributes and methods to classes Deadline and Event

Progresst-8 committed:
Add LocalDate attributes and in Construction of Deadline and Event objects

Progresst-8 committed:
Add date and time formatting to Deadline and Event classes

Amend overriden taskToDisplay and taskToString methods in both Deadline and Evenet classes
Deadline and Event classes share similar date constructors and require the same string format constants and getDateTimeFromString
Add new class DateAndTimeHandler to store these shared attributes and methods

Progresst-8 committed:
Add find functionality to Yapper chatbot program, execution-wise (not parsing-wise)

FIND lists all matching tasks (not just the top N), is case-sensitive (not case-insensitive), match substring (not match whole words) and searches description only (not start date or end date).
Add printTaskWithoutOrdinal and printTaskWithOrdinal methods to OutputStringHandler to adhere to SLAP, and also because similar patterns of code is found throughout OutputStringHandler
printAllTasks in OutputStringHandler used to use .get() which is a ArrayList method and not a List<Task> tasks method, so TaskManager is passed as an argument so that I can use getTaskOrdinal() or getTask() instead
Add printTasksMatchQuery method in OutputStringHandler to handle FIND, which follows the idea above
Edit printAllTasks and printTasksMatchQuery function calls in handleFindInstruction and handleListInstruction (in InstructionHandler) respectively
Add in FIND in handleInstruction method's match-case and a handleFindInstruction
Add FIND to InstructionType enum
Add 2 string constants related to finding in StringStorage class

Progresst-8 committed:
Change output messages for FIND

Add string constants to StringStorage
Add comments in InstructionHandler

Progresst-8 committed:
Add personality to Yapper's yappin

Every print method in OutputStringHandler now has a dedicated string before and after it.
Add string constants to StringStorage class

Progresst-8 committed:
Rename methods

For consistency and clarity, the 4 print methods start with "print" and the 2 methods that don't print (but instead returns a String) start with "display"

Progresst-8 committed:
Remove extra LINE_DIVIDER, else Yapper prints 2 when user inputs BYE

Progresst-8 committed:
Add find functionality to Yapper chatbot program, parsing-wise

Update checkIfStartWithInstructionPrefix method to also check for FIND (in ExceptionHandler)
Add exception checker method checkIfFindArgsMissing (in ExceptionHandler)
Add FIND switch-case branch in parseUserInput method (in InputStringHandler class)
Add string constants related to finding (in StringStorage class)
Add comment in Instruction class

Progresst-8 committed:
Fix bug where printAllTasks and printTasksMatchQuery did not print to output

2 print methods in OutputStringHandler class did not print tasks because display methods returns a string and does not print.
Rename 2 TaskHandler methods for clarity
Add newline to RELEVANT_TASKS_FOUND_STRING

Progresst-8 committed:
Made Yapper more conversational and other minor changes

Change string constants in StringStorage class that are used in OutputStringHandler class
Fix syntax errors in OutputStringHandler class
Rename printTasksMatchQuery method in OutputStringHandler class as it was too wordy

Progresst-8 committed:
Merge pull request #1 from Progresst-8/branch-Level-8

Add date and time functionality to Yapper

Progresst-8 committed:
Merge branch 'master' of https://github.com/Progresst-8/ip

Progresst-8 committed:
Merge branch 'master' into branch-Level-9

Progresst-8 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/yapper/tasks/Deadline.java
#	src/main/java/yapper/tasks/Event.java

Progresst-8 committed:
Made formatting changes

Progresst-8 committed:
Merge pull request #2 from Progresst-8/branch-Level-9

Merging branch-level-9 into master

Progresst-8 committed:
Merge branch 'master' of https://github.com/Progresst-8/ip

Progresst-8 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/yapper/exceptions/ExceptionHandler.java
#	src/main/java/yapper/instructions/InstructionHandler.java
#	src/main/java/yapper/io/OutputStringHandler.java
#	src/main/java/yapper/io/StringStorage.java
#	src/main/java/yapper/tasks/TaskHandler.java

Progresst-8 committed:
Merge branch 'branch-A-JavaDoc'

Progresst-8 committed:
Add support for a HELP command that prints some instructions.

Progresst-8 committed:
Edited 2 README.md files

One is a quickstart, the other (in a docs folder) is a User Guide

Progresst-8 committed:
Remove an unintended extra space when printing Deadlines or Events

Progresst-8 committed:
Change some comments in StringStorage

Progresst-8 committed:
Made changes to User Guide README

Progresst-8 committed:
Clean up code, add some JavaDoc

Progresst-8 committed:
Fix bug when parsing data from text file

Progresst-8 committed:
Add javadocs

Progresst-8 committed:
Swapped class names for Input and Output File Handling, as they were misnamed

Progresst-8 committed:
Add test class files

Progresst-8 committed:
Create DG, update UG

Progresst-8 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Progresst-8 committed:
Add checkstyle files according to SE website

Progresst-8 committed:
Add test code classes

okkhoy commented on .gitignore:
```diff
@@ -15,3 +15,21 @@ bin/
 
 /text-ui-test/ACTUAL.TXT
 text-ui-test/EXPECTED-UNIX.TXT
+<<<<<<< Updated upstream
+=======
+src/main/java/yapper/CommonStrings.class
+src/main/java/yapper/io/InputStringHandler.class
+src/main/java/Instruction$InstructionType.class
+src/main/java/yapper/instructions/Instruction.class
+src/main/java/yapper/io/OutputStringHandler.class
+src/main/java/yapper/io/StringStorage.class
+src/main/java/yapper/tasks/Task.class
+src/main/java/yapper/tasks/TaskDeadline.class
+src/main/java/yapper/tasks/TaskEvent.class
+src/main/java/yapper/TaskManager.class
+src/main/java/yapper/tasks/TaskTodo.class
+src/main/java/UserInterface$1.class
+src/main/java/yapper/instructions/InstructionHandler.class
+src/main/java/yapper/Yapper.class
+>>>>>>> Stashed changes
```
why are you committing stashed changes/merge conflicts?

okkhoy commented on src/main/java/yapper/Yapper.java:
```diff
@@ -0,0 +1,84 @@
+     *
+     * @param taskHandler the TaskHandler instance that manages tasks
+     */
+    public static void startYappin(TaskHandler taskHandler) {
```
while the name is "cool", for someone maintaining the code, what does `startYappin` mean? use a name that represents the operation

okkhoy commented on src/main/java/yapper/Yapper.java:
```diff
@@ -0,0 +1,84 @@
+     * @param args command line arguments (not used)
+     */
+    public static void main(String[] args) {
+        // Pre-Program Message
```
looks like a misleading comment

okkhoy commented on src/main/java/yapper/exceptions/ExceptionHandler.java:
```diff
@@ -0,0 +1,231 @@
+ */
+public class ExceptionHandler {
+
+    // User Input Validation Errors
```
redundant comment. it doesn't refer to any code immediately.

okkhoy commented on src/main/java/yapper/exceptions/ExceptionHandler.java:
```diff
@@ -0,0 +1,231 @@
+     * @param isAboutArgs flag indicating if the check is for arguments
+     * @throws YapperException if the input string is empty
+     */
+    public static void checkIfUserInputEmpty(String userInputString, boolean isAboutArgs) throws YapperException {
```
can this be 2 separate methods - one while reading the file and the other while reading input from user?

okkhoy commented on src/main/java/yapper/exceptions/ExceptionHandler.java:
```diff
@@ -0,0 +1,231 @@
+        }
+    }
+
+}
```
can consider splitting it to 2 classes one for reading data from storage other for managing user inputs

okkhoy commented on src/main/java/yapper/instructions/Instruction.java:
```diff
@@ -0,0 +1,190 @@
+    }
+
+
+    // Getters
```
redundant.

okkhoy commented on src/main/java/yapper/instructions/InstructionHandler.java:
```diff
@@ -0,0 +1,207 @@
+     * @throws YapperException If an error occurs while deleting the task.
+     */
+    public static void handleDeleteInstruction(TaskHandler taskHandler, Integer taskOrdinal) throws YapperException {
+        // OOB method should indirectly check if list is empty?
```
looks like a "note to self" comment; if so, should be removed

okkhoy commented on src/main/java/yapper/io/InputFileHandler.java:
```diff
@@ -0,0 +1,154 @@
+            // Check Desc and other Arguments, indirectly checks if missing
+            String remainingParts = taskParts[2].trim();
+            switch (taskType) {
+                case StringStorage.TODO_SYMBOL:
```
coding standard violation

okkhoy commented on src/main/java/yapper/io/StringStorage.java:
```diff
@@ -0,0 +1,280 @@
+package yapper.io;
+
+// TODO: make more conversational
```
is it done? if so remove the comment.

okkhoy commented on src/main/java/yapper/io/StringStorage.java:
```diff
@@ -0,0 +1,280 @@
+    /**
+     * Constants representing prefixes for user commands, for User I/O.
+     */
+    public static final String     HELP_INSTRUCTION_PREFIX = "help";
+    public static final String     FIND_INSTRUCTION_PREFIX = "find";
+    public static final String     LIST_INSTRUCTION_PREFIX = "list";
+    public static final String     TODO_INSTRUCTION_PREFIX = "todo";
+    public static final String DEADLINE_INSTRUCTION_PREFIX = "deadline";
+    public static final String    EVENT_INSTRUCTION_PREFIX = "event";
+    public static final String   DELETE_INSTRUCTION_PREFIX = "delete";
+    public static final String     MARK_INSTRUCTION_PREFIX = "mark";
+    public static final String   UNMARK_INSTRUCTION_PREFIX = "unmark";
+    public static final String      BYE_INSTRUCTION_PREFIX = "bye";
```
better PREFIX_xxxxxxxxx

okkhoy commented on src/main/java/yapper/tasks/Event.java:
```diff
@@ -0,0 +1,180 @@
+     * @param endDateString the end date string to be parsed
+     */
+    private void initializeEndDateTime(String endDateString) {
+        try {
+            this.endDateTime = LocalDateTime.parse(endDateString,
+                    DateTimeFormatter.ofPattern(
+                    DateAndTimeHandler.DATE_WITH_TIME_INPUT) );
+            this.endDateString = "";
+        } catch (DateTimeParseException e) {
+            this.endDateString = "";
+            try {
+                this.endDate = LocalDate.parse(endDateString,
+                        DateTimeFormatter.ofPattern(
+                        DateAndTimeHandler.DATE_WITHOUT_TIME_INPUT) );
+            } catch (DateTimeParseException ex) {
+                this.endDateString = endDateString;
+            }
+        }
```
think if there is a way to simplify this using some `if...else` checks

okkhoy commented on src/test/java/testyapper/testinstructions/TestInstructionHandler.java:
```diff
@@ -0,0 +1,8 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestInstructionHandler {
+    ;
```
???

okkhoy commented on src/test/java/testyapper/testio/TestDateAndTimeHandler.java:
```diff
@@ -0,0 +1,6 @@
+
+
+public class TestDateAndTimeHandler {
+    ;
```
??
don't commit if not completed!

okkhoy reviewed (commented):
You have too many redundant comments in the code!

Progresst-8 committed:
Made changes according to CheckStyle requirements

Add JavaDoc for class DateAndTimeHandler
Add newline to end of file, ensure line length does not exceed a certain length, arranged import order lexicographically, avoid star import, add default switch case, operator wrap, whitespace removal

Progresst-8 committed:
Rename PREFIX, SYMBOL, DELIMITER constants according to tutor's suggestion

Progresst-8 committed:
Corrected name of constant

Progresst-8 committed:
Change program output for exception catching and for input file handling

Progresst-8 committed:
Corrected name of constant in InputStringHandler

Progresst-8 committed:
Clean up code

Progresst-8 committed:
Extract repeated code into an exception handling method that checks for arguments for instructions that don't require them

Progresst-8 committed:
Move method to reflect the order it is used in parseUserInput

Progresst-8 committed:
Fix markdown table

Progresst-8 committed:
Remove comments as suggested by tutor.

Add JavaDoc to replace certain comments

Progresst-8 committed:
Fix code that violated checkstyle

Progresst-8 committed:
Removed unused method setDesc in Task class

Progresst-8 committed:
Update User Guide README

Add Find to User Guide

Progresst-8 committed:
Update User Guide

Progresst-8 committed:
Update User Guide

Progresst-8 committed:
Remove comments

Progresst-8 committed:
Update DG

Progresst-8 committed:
Update TASK_IS_DONE string

Progresst-8 committed:
Reformat some code

Progresst-8 committed:
Reduce reliance on List by rewriting code

Renamed variable

Progresst-8 committed:
Update setup README

Progresst-8 committed:
Remove comments

Progresst-8 committed:
Update JavaDoc

Progresst-8 committed:
Rename method for clarity

Progresst-8 committed:
Rename method for clarity

Progresst-8 committed:
Rewrote code

Refactor parsing code out of handleInstruction method into its own
Main loop now calls parseUserInput and instructionHandler instead of just calling instructionHandler (which internally calls parseUserInput)
Redid the way exception messages are formed, what parseUserInput used to do is now done by parseAndValidateUserInput
Add JavaDocs where necessary

Progresst-8 committed:
Change JavaDoc

Progresst-8 committed:
Update DG

Progresst-8 committed:
Move .trim() to when scanning in string

Progresst-8 committed:
Update code to not need printWithDividers method

Progresst-8 committed:
Update JavaDoc

Progresst-8 committed:
Refactored repeated code that converts between ArrayList and File

Progresst-8 committed:
Add exception handling for FileNotFoundException

Progresst-8 committed:
Rewrote how loadTasks method in InputFileHandler class worked

Progresst-8 committed:
Change access modifiers of helper methods from public to private

Progresst-8 committed:
Add exception messages for default switch-case branch

Progresst-8 committed:
Remove code that ends scanning file if blank line is detected

Progresst-8 committed:
Update UG to mention file saving and retrieval

Progresst-8 committed:
Merge branch 'master' into streamlineFileHandling

# Conflicts:
#	src/main/java/yapper/io/InputFileHandler.java

Progresst-8 committed:
Add JavaDoc to the sole taskHandler that the program uses

Progresst-8 committed:
Improve exception handling in storeAddedTask method

Progresst-8 committed:
Remove redundant end scanning and put newlines for clarity

Fix bug causing taskHandler to be null, before performing List operations on it. Initializing FileHandler will initialize a TaskHandler, solving this issue.

Progresst-8 committed:
Change helper methods to private

Progresst-8 committed:
Add method to convert task list to file format

Progresst-8 committed:
Merge branch 'streamlineFileHandling'

Progresst-8 committed:
Add method to convert all tasks in the task list to strings

Progresst-8 committed:
Add initializing and isEmpty methods to FileHandler class

Progresst-8 committed:
Refactor out tasksToString code from OutputFileHandler to TaskHandler

Progresst-8 committed:
Add code to check for save folder and save file during runtime

Progresst-8 committed:
Remove JUnit test classes

Progresst-8 committed:
Update UG to reflect current file saving capabilities

Progresst-8 committed:
Update UG

Progresst-8 committed:
Check for save file and folder is now done in Yapper, not handleInstruction

Remove unecessary comments

Progresst-8 committed:
Validation for save file and folder is now done in Yapper.

Instead of checking only after parsing an instruction without encountering exceptions, it now checks even if parsing throws an exception

Progresst-8 committed:
Add string to StringStorage

Progresst-8 committed:
Add string to StringStorage

Progresst-8 committed:
Factor out strings to StringStorage

Progresst-8 committed:
Rework how initializing save folder and save file worked

Progresst-8 committed:
Update INPUT and ACTUAL file

Progresst-8 committed:
Change where file validation method is in main loop

Progresst-8 committed:
Remove unnecessary class access

Progresst-8 committed:
Update UG

Progresst-8 committed:
Reword UG
